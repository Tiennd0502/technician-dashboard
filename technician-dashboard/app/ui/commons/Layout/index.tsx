'use client';

import { FC, ReactNode, useMemo } from 'react';
import { Grid, GridItem, useMediaQuery } from '@chakra-ui/react';

import { MENU_ITEM_LIST, HEADER_INFO } from '@/lib/constants';

import { Sidebar, Header, MiniSidebar } from '..';

const GRID_TEMPLATE = `
  'nav header'
  'nav main'
`;

const GRID_SMALL_TEMPLATE = `
  'nav'
  'header'
  'main'
`;

const Layout: FC<{ children: ReactNode }> = ({ children }) => {
  const [isSmallScreen] = useMediaQuery('(max-width: 768px)');

  const styledWrapper = useMemo(
    () =>
      isSmallScreen
        ? {
            templateAreas: GRID_SMALL_TEMPLATE,
            gridTemplateColumns: '100%',
          }
        : {
            templateAreas: GRID_TEMPLATE,
            gridTemplateRows: '100px 1fr',
            gridTemplateColumns: '250px 1fr',
          },
    [isSmallScreen],
  );

  return (
    <Grid gap='5' {...styledWrapper}>
      <GridItem area='nav'>
        {isSmallScreen ? (
          <MiniSidebar listItem={MENU_ITEM_LIST} />
        ) : (
          <Sidebar listItem={MENU_ITEM_LIST} />
        )}
      </GridItem>
      <GridItem area='header'>
        <Header {...HEADER_INFO} />
      </GridItem>
      <GridItem area='main'>{children}</GridItem>
    </Grid>
  );
};

export default Layout;
