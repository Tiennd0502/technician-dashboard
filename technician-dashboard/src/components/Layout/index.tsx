import { FC, ReactNode } from 'react';
import { Grid, GridItem } from '@chakra-ui/react';

import { MENU_ITEM_LIST, HEADER_INFO } from '@/constants';

import { Sidebar, Header } from '..';

const GRID_TEMPLATE = `
  "nav header"
  "nav main"
`;

const Layout: FC<{ children: ReactNode }> = ({ children }) => (
  <Grid
    gap='5'
    templateAreas={GRID_TEMPLATE}
    gridTemplateRows='100px 1fr'
    gridTemplateColumns='250px 1fr'
  >
    <GridItem area='nav'>
      <Sidebar listItem={MENU_ITEM_LIST} />
    </GridItem>
    <GridItem area='header'>
      <Header {...HEADER_INFO} />
    </GridItem>
    <GridItem area='main'>{children}</GridItem>
  </Grid>
);

export default Layout;
