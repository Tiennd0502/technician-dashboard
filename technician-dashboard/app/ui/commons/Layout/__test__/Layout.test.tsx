import { render } from '@testing-library/react';
import Layout from '..';

describe('Layout component', () => {
  const props = {
    children: 'Children',
  };

  test('should show children', () => {
    const { getByText } = render(<Layout {...props} />);
    const children = getByText('Children');

    expect(children).toBeTruthy();
  });

  test('match snapshot', () => {
    const { container } = render(<Layout {...props} />);

    expect(container).toMatchSnapshot();
  });
});
