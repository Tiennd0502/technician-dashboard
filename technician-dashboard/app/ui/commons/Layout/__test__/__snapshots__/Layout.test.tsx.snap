// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component match snapshot 1`] = `
<div>
  <div
    class="css-sec3a1"
  >
    <div
      class="css-1o29zam"
    >
      <div
        class="css-1hrbm4s"
      >
        <div
          class="css-150u52e"
        >
          <div
            class="css-1owjrhv"
          >
            <img
              alt="Technician"
              data-nimg="1"
              decoding="async"
              height="38"
              loading="lazy"
              src="/images/technician.svg"
              style="color: transparent;"
              width="37"
            />
            <div
              class="css-0"
            >
              <p
                class="chakra-text css-p3xeyl"
              >
                Technician
              </p>
              <p
                class="chakra-text css-1o9nuri"
              >
                At Home
              </p>
            </div>
          </div>
          <button
            aria-label="Icon Toggle Menu"
            class="chakra-button css-4xx2wk"
            type="button"
          >
            <svg
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="css-ubyy1i"
    >
      <div
        class="css-1a3oakf"
      >
        <h1
          class="chakra-heading css-0"
        >
          Product & Services
        </h1>
        <div
          class="css-147dy5z"
        >
          <div
            class="css-1ve4u7y"
            data-testid="icon-button-component"
          >
            <div
              class="css-u3hjaz"
            />
            <button
              aria-label="Notifications"
              class="chakra-button css-p7mkev"
              data-testid="icon-button"
              type="button"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 23 24"
                width="23"
              >
                <path
                  d="M5.45605 15.9237L4.58934 15.4249C4.41126 15.7343 4.41165 16.1152 4.59037 16.4243C4.7691 16.7334 5.09904 16.9237 5.45605 16.9237V15.9237ZM18.4422 15.9237V16.9237C18.7992 16.9237 19.1292 16.7334 19.3079 16.4243C19.4866 16.1152 19.487 15.7343 19.3089 15.4249L18.4422 15.9237ZM21.8983 15.9237H22.8983C22.8983 15.3714 22.4505 14.9237 21.8983 14.9237V15.9237ZM3.98474 8.96439H2.98474H3.98474ZM21.8983 17.9237V18.9237C22.4505 18.9237 22.8983 18.476 22.8983 17.9237H21.8983ZM9.3614 20.4042L8.49486 19.9051L9.3614 20.4042ZM14.5369 20.4042L13.6703 20.9033H13.6703L14.5369 20.4042ZM14.5353 21.4051L13.6703 20.9033H13.6703L14.5353 21.4051ZM9.36293 21.4051L8.49793 21.9069L9.36293 21.4051ZM6.98474 8.96439C6.98474 6.22263 9.20737 4 11.9491 4V2C8.1028 2 4.98474 5.11806 4.98474 8.96439H6.98474ZM6.98474 13.939V8.96439H4.98474V13.939H6.98474ZM6.32277 16.4225C6.74408 15.6904 6.98474 14.8413 6.98474 13.939H4.98474C4.98474 14.4819 4.84067 14.9882 4.58934 15.4249L6.32277 16.4225ZM18.4422 14.9237H5.45605V16.9237H18.4422V14.9237ZM16.9135 13.939C16.9135 14.8413 17.1542 15.6904 17.5755 16.4225L19.3089 15.4249C19.0576 14.9882 18.9135 14.4819 18.9135 13.939H16.9135ZM16.9135 8.96439V13.939H18.9135V8.96439H16.9135ZM11.9491 4C14.6909 4 16.9135 6.22263 16.9135 8.96439H18.9135C18.9135 5.11806 15.7955 2 11.9491 2V4ZM21.8983 14.9237C21.3544 14.9237 20.9135 14.4828 20.9135 13.939H18.9135C18.9135 15.5874 20.2498 16.9237 21.8983 16.9237V14.9237ZM20.9135 13.939V8.96439H18.9135V13.939H20.9135ZM20.9135 8.96439C20.9135 4.01349 16.9 0 11.9491 0V2C15.7955 2 18.9135 5.11806 18.9135 8.96439H20.9135ZM11.9491 0C6.99823 0 2.98474 4.01349 2.98474 8.96439H4.98474C4.98474 5.11806 8.1028 2 11.9491 2V0ZM2.98474 8.96439V13.939H4.98474V8.96439H2.98474ZM2.98474 13.939C2.98474 14.4828 2.54386 14.9237 2 14.9237V16.9237C3.64843 16.9237 4.98474 15.5874 4.98474 13.939H2.98474ZM2 14.9237C0.895431 14.9237 0 15.8191 0 16.9237H2V14.9237ZM0 16.9237C0 18.0283 0.895431 18.9237 2 18.9237V16.9237H0ZM2 18.9237H21.8983V16.9237H2V18.9237ZM22.8983 17.9237V15.9237H20.8983V17.9237H22.8983ZM10.2279 18.9033C9.51267 18.9033 8.85186 19.2853 8.49486 19.9051L10.2279 20.9033V18.9033ZM13.6703 18.9033H10.2279V20.9033H13.6703V18.9033ZM15.4034 19.9051C15.0464 19.2853 14.3856 18.9033 13.6703 18.9033V20.9033H13.6703L15.4034 19.9051ZM15.4003 21.9069C15.7592 21.2882 15.7604 20.5249 15.4034 19.9051L13.6703 20.9033L15.4003 21.9069ZM11.9491 23.8947C13.3716 23.8947 14.6866 23.1374 15.4003 21.9069L13.6703 20.9033C13.3144 21.517 12.6586 21.8947 11.9491 21.8947V23.8947ZM8.49793 21.9069C9.21171 23.1374 10.5266 23.8947 11.9491 23.8947V21.8947C11.2397 21.8947 10.5839 21.517 10.2279 20.9033L8.49793 21.9069ZM8.49486 19.9051C8.13786 20.5249 8.13903 21.2882 8.49793 21.9069L10.2279 20.9033L8.49486 19.9051Z"
                  fill="#4c52bc"
                />
              </svg>
            </button>
          </div>
          <button
            aria-controls="menu-list-:rb:"
            aria-expanded="false"
            aria-haspopup="menu"
            class="chakra-button chakra-menu__menu-button css-4xx2wk"
            data-testid="btn-dropdown"
            id="menu-button-:rb:"
            type="button"
          >
            <span
              class="css-xl71ch"
            >
              <div
                class="css-guqklo"
              >
                <span
                  aria-label="Avatar"
                  class="chakra-avatar css-1ruhx4w"
                >
                  <div
                    aria-label="Avatar"
                    class="chakra-avatar__initials css-0"
                    role="img"
                  >
                    A
                  </div>
                </span>
                <p
                  class="chakra-text css-0"
                >
                  Belle
                </p>
              </div>
            </span>
            <span
              class="chakra-button__icon css-1hzyiq5"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-q5208i"
                data-testid="down-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
          <div
            class="css-0"
            style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
          >
            <div
              aria-orientation="vertical"
              class="chakra-menu__menu-list css-sjt5nk"
              data-testid="dropdown-options"
              id="menu-list-:rb:"
              role="menu"
              style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
              tabindex="-1"
            >
              <button
                class="chakra-menu__menuitem css-1k756t9"
                data-index="0"
                id="menu-list-:rb:-menuitem-:rc:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                Profile
              </button>
              <button
                class="chakra-menu__menuitem css-1k756t9"
                data-index="1"
                id="menu-list-:rb:-menuitem-:rd:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                Change password
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-qfa1zt"
    >
      Children
    </div>
  </div>
</div>
`;
